src/engine/sync/basesyncablelistener.cpp

```diff
@@ -124,22 +124,24 @@ void BaseSyncableListener::setMasterBeatDistance(Syncable* pSource, double beatD
     }
 }
 
-void BaseSyncableListener::setMasterParams(
-        Syncable* pSource, double beatDistance, double baseBpm, double bpm) {
-    //qDebug() << "BaseSyncableListener::setMasterParams, source is" << pSource->getGroup() << beatDistance << baseBpm << bpm;
+void BaseSyncableListener::setMasterParams(Syncable* pSource) {
+    const double beatDistance = pSource->getBeatDistance();
+    const double baseBpm = pSource->getBaseBpm();
+    const double bpm = pSource->getBpm() > 0 ? pSource->getBpm() : pSource->getBaseBpm();
```

Comment:
This assumes the BPM value doesn't change between the comparison and the true case. In my midi clock implementation this will be atomic and may change at any time. Maybe we should just store it in a variable first, and do the comparison on that value?

Reply:
sure